py/picca/fitter/cosmo.py

```diff
@@ -234,9 +270,11 @@ def getXiAuto(self,rp,rt,z,pk_lin,pars):
         sigmaNLper = pars["SigmaNL_perp"]
         sigmaNLpar = sigmaNLper*pars["1+f"]
         pk_nl = sp.exp(-(kp*sigmaNLpar)**2/2-(kt*sigmaNLper)**2/2)
-        pk_full*=pk_nl
-        if self.dnl :
-            pk_full*=self.DNL(k,muk)
+        pk_full *= pk_nl
+        if self.dnl_mcdonald :
+            pk_full *= self.DNL(k,muk,self.pk,self.q1_dnl,self.kv_dnl,self.av_dnl,self.bv_dnl,self.kp_dnl,"mcdonald")
```

Comment:
here and in other places, wouldn't it be better to remove this if and call the function DNL with "self.dnl_method" as an argument?

Reply:
self.dnl_method is just a bool parameter, so one would have to pass two arguments: self.dnl_mcdonald and self.dnl_arinyo. I can do that.

Reply:
you could simplify this by setting self.dnl_method to None by default and:

`
if self.dnl_method is not None:
    apply DNL(...,self.dnl_method)

Reply:
and upgrade self.dnl_method to a string